### PR TITLE
refactor(generators): remove extraneous args to addon-generator

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**
refactoring

**Did you add tests for your changes?**
Already present

**If relevant, did you update the documentation?**
N/A

**Summary**
https://github.com/webpack/webpack-cli/blob/65ae77d98e918ea5907de89c9c87b797f57f1b2f/packages/generators/src/addon-generator.ts#L30

Directory content is inferred from the `templateDir` argument. Hence, it is not necessary to explicitly specify the template files.

**Does this PR introduce a breaking change?**
No

**Other information**
N/A

### CODE REVIEW

**Review by alexander-akait — DISMISSED** [review]
/cc @webpack/cli-team

File: packages/generators/src/addon-generator.ts [hunk 1]
```diff
@@ -4,6 +4,16 @@ import Generator from 'yeoman-generator';
 
 import { List } from './utils/scaffold-utils';
 
+// Helper to get the template-directory content
+
+const getFiles = (dir) => {
+    return fs.readdirSync(dir).reduce((list, file) => {
+        const filePath = path.join(dir, file);
+        const isDir = fs.statSync(filePath).isDirectory();
```
Comment (alexander-akait) [inline packages/generators/src/addon-generator.ts:12]: Maybe we need handle this case better, in some edge cases here can be exception, for example permissions

Comment (jamesgeorge007) [reply packages/generators/src/addon-generator.ts:12]: How do you see it? In fact, this is used by projects like [vite](https://github.com/vitejs/vite)
https://github.com/vitejs/vite/blob/978d991293f5b8e1a4197ac4e3aee4fa2e838a88/packages/create-app/index.js#L318

Comment (alexander-akait) [reply packages/generators/src/addon-generator.ts:12]: Yep, used, but not handle exceptions, just add `try/catch` around `getFiles`, i.e. `try { getFiles() } catch (error) { logger.error(error); process.exit(2) }`

File: packages/generators/src/addon-generator.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import fs from 'fs';
 import path from 'path';
+import createLogger from 'yeoman-environment/lib/util/log';
```
Comment (alexander-akait) [inline packages/generators/src/addon-generator.ts:3]: Let's avoid it,w e ahve built-in logger^^

Comment (jamesgeorge007) [reply packages/generators/src/addon-generator.ts]: Oops

**Review by alexander-akait — APPROVED** [review]

**Review by snitin315 — APPROVED** [review]

**Pull request merged**